src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs

```diff
@@ -28,35 +28,36 @@
 
 using System.Collections.Generic;
 using System.Diagnostics;
-using System.Runtime.Serialization;
 using System.Threading;
 
-#if ENABLE_WINRT
-using Internal.Runtime.Augments;
-#endif
-
 namespace System.Globalization
 {
+#if CORECLR
+    using StringCultureInfoDictionary = Dictionary<string, CultureInfo>;
+    using StringLcidDictionary = Dictionary<int, CultureInfo>;
+    
+    using Lock = Object;
+#else
+    using StringCultureInfoDictionary = LowLevelDictionary<string, CultureInfo>;
+    using StringLcidDictionary = LowLevelDictionary<int, CultureInfo>;
+#endif
```

Comment:
Apparently this part is problematic because in the `!CORECLR` case `LockHolder` is not properly used. I'd appreciate suggestions on the preferred way to fix this.

Reply:
I would change this to use regular Dictionary everywhere. The `LowLevelDictionary` optimization has dubious value since 95% of framework is using regular `Dictionary`. One more or one less won't make big difference.